core/src/main/scala/scalaz/Ap.scala

```diff
@@ -0,0 +1,41 @@
+package scalaz
+
+/** Derive a Semigroup or Monoid instance from an Apply or Applicative.
+ *
+ * https://hackage.haskell.org/package/reducers-3.12.1/docs/Data-Semigroup-Applicative.html#t:Ap
+ */
+case class Ap[F[_], A](f: F[A])
```

Comment:
Why not `final case class`?
I'm not sure whether scalaz has a policy on this or not.